### PR TITLE
HYB-695: Deeplinking

### PR DESCRIPTION
**Pull request by lizcross**
Enable universal deep linking for ios

JIRA: https://mobify.atlassian.net/browse/HYB-695
Linked PRs: https://github.com/mobify/generator-astro/pull/26

## Changes
- Add `continueUserActivity` in `AppDelegate`
- Add an entitlements file to build
- Add a `apple-app-site-association` which can be further configured and hosted on the deep linked site

## How to test-drive this PR
- Run the generator and ensure that the entitlements file and association file are updated correctly

## TODOS:
- [ ] Change works in both ~~Android and~~ iOS.
- [ ] CHANGELOG.md has been updated.
- [ ] Scaffold is working. If a new feature was added, it was added to the Scaffold app.
- [ ] Run the generator to make sure it still functions correctly.
- [ ] +1 from an engineer on the Astro team.
- [ ] Both tab layout and drawer layout are fully functional in ios. (Set `iosUsingTabLayout` in `baseConfig`)

### CODE REVIEW

**Comment by tony-dinh** [comment]
I am not too familiar with setting up a project with universal deep linking so I am mostly looking at diffs between the merged branch on [Matalan](https://github.com/mobify/matalan/commit/f5ee86f30dc2773a659c89994ef9bde27aa52442) and this one.

I do not see `scaffold.entitlements` like I do with `matalan.entitlements`... is that an issue? I can see that it has been added to the build settings and it is looking for scaffold/scaffold.entitlements which is exists in the finder but not bundled in the app. Not sure if it was intentionally left out

**Comment by lizcross** [comment]
@tony-dinh good points. I was not entirely sure how to setup this file in the build. @jeremywiebe could you take a look and see if the changes Tony suggested are necessary?

File: ios/scaffold/scaffold.entitlements [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>com.apple.developer.associated-domains</key>
+	<array>
+		<string>applinks:www.mobify.com</string>
```
Comment (jansepar) [inline ios/scaffold/scaffold.entitlements]: Should this be velo?

**Comment by kschmidtdev** [comment]
I think this can be closed in lieu of #106

File: ios/scaffold/apple-app-site-association [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (jeremywiebe) [inline ios/scaffold/apple-app-site-association]: Can we add a reference to this in the README of the Scaffold?  Even if it's just a pointer to our Astro docs that deal with deep linking just so somebody generating an app has a place to start from (too bad we can't put comments in this file... but alas... JSON).

**Comment by lizcross** [comment]
Creating a new PR to just add the apple-app-site-association file.

**Comment by kerrmarin** [comment]
We're closing this since it hasn't had any activity in the past couple of months and seems to have been superseded by #106

**Pull request closed without merging**